### ISSUE TITLE
Create Simple Segmentation network

### ISSUE THREAD

**Issue by tomassosorio** [tomassosorio/DeepLearning#31]
Use example simple object detection, and as GT change it with an matrix with 0 representing background 1 as object.
Only using conv layers.

**Comment by tomassosorio** [comment]
Proposal:

We only will need to use one class since our objective is also only one class.

Create an 2D matrix with the GT as label.

To train use CNN and the ouput will have to be [img_size, img_size] x 1 channel, with the confidence of each pixel be active or not.

**Comment by tomassosorio** [comment]
https://github.com/tomassosorio/DeepLearning/blob/dev/Simple%20Segmentation%20Network/single-rectangle-segmentation.ipynb

Basic Concept is working

**Comment by miguelriemoliveira** [comment]
Did not understand very well what you did here differing from your initially proposed approach

> To train use CNN and the ouput will have to be [img_size, img_size] x 1 channel, with the confidence of each pixel be active or not.

 Can you explain?

**Comment by tomassosorio** [comment]
What I did was:
Create the ground truth (GT) simultaneously with the generation of the images.

In the example that I did I reshaped the GT to an array with 1D, that will be compared with the output of the softmax layer (last layer), using a simple MLP, standard neural networks.

In the proposed approach, I tough about using CNN instead of MLP. Using some initial convolution layers, and as the last layer have an output of [size x size] x 1 channel, comparing after with the GT, (I am not sure if this is possible) and calculate the loss, this way avoiding the usage of standard layers.

**Comment by miguelriemoliveira** [comment]
I may be wrong, but the spatial or pixel neighborhood information would be very important (if not crucial) for the segmentation of which pixels belong to the road. I mean, if its purely based on color of the pixel of course not, but a more effective classifier would have to take into account the connectivity of the pixels.

With a 1D array as input and a classical multi layer perceptron this is not possible right?

Note that one of the first applications of neural networks to the autonomous driving area was the navlab. You should check these links (out of curiosity, but also because its important to know were things have started):

http://repository.cmu.edu/cgi/viewcontent.cgi?article=2874&context=compsci

https://www.ri.cmu.edu/pub_files/pub3/pomerleau_dean_1991_1/pomerleau_dean_1991_1.pdf

**Comment by tomassosorio** [comment]
You are absolutely right @miguelriemoliveira , spatial or pixel neighborhood information is very important, but the idea here was just to prove the concept and that we can do.
In a next phase, we can increase the difficulty by using convolution layers.
Thank you for the links.
